paddle/fluid/platform/profiler.cc

```diff
@@ -173,8 +173,9 @@ void PopEvent(const std::string& name, const DeviceContext* dev_ctx) {
 
 RecordEvent::RecordEvent(const std::string& name, const DeviceContext* dev_ctx)
     : is_enabled_(false), start_ns_(PosixInNsec()) {
-  std::lock_guard<std::mutex> l(profiler_mu);
   if (g_state == ProfilerState::kDisabled) return;
+  std::lock_guard<std::mutex> l(profiler_mu);
```

Comment:
Does this matter? it's wrong to put it here.

Reply:
Maybe don't need to check the mutex when disabling the profiler, it will decrease the performance, and disable/enable the profiler only happens at the beginning or end of each batch training.